CDP4Dal/Assembler.cs

```diff
@@ -124,13 +130,17 @@ public async Task Synchronize(IEnumerable<CDP4Common.DTO.Thing> dtoThings, bool
             }
 
             await this.threadLock.WaitAsync().ConfigureAwait(false);
+
             try
             {
                 var synchronizeStopWatch = Stopwatch.StartNew();
 
                 logger.Info("Start Synchronization of {0}", this.IDalUri);
 
+                this.olderRevisionDtos = new List<Dto>();

```

Comment:
not sure this needs to be a private field. It is initialized in this method, passed to the `UpdateThingRevisions` , altered there and then used in the calling method to filter out data... i propose that you remove it as private field from the class and pass it as `ref`  argument to `UpdateThingRevisions`, the `ref`  making it clear we are changing the argument, or at least it's content

Reply:
Initially had set it up that way.
I changed it to conform to the implementation of thingsMarkedForDeletion.
I'll change it back.